src/Model/Subscription.php

```diff
@@ -113,6 +113,33 @@ public function setRemainingValue(Money $value): self
         return $this;
     }
 
+    /**
+     * Based off of the current Price, Modifiers, and BillingPeriod on the
+     * Subscription, calculate how much of the Subscription's total cost
+     * remains. When a subscription is fetched from provider the response of this
+     * is passed to ->setRemainingValue(...) so that any modifications to Subscription
+     * attributes do not change the remaining value when ->getRemainingValue() is called.
+     * However, if you need to recalculate the remaining value, you can do it right
+     * from the Subscription itself.
+     */
+    public function calculateRemainingValue(): Money
```

Comment:
Can you explain when you would use `calculateRemainingValue` instead of `getRemainingValue`. The comment sheds a little light, but I'm not certain.

Would you use `calculateRemainingValue()` when you fetch something from the server and then augment modifiers?

What is the signal to an end user to reach for this method instead of `getRemainingValue()`?

Reply:
After looking at the tests and the mapper, I think I understand.

The naming is a pretty confusing though. Maybe we just roll with it for now.

I'm not saying we do this, but maybe it will help solidify my understanding. If `Subscription` were immutable, would we need `calculateRemainingValue()`?

```php
$subscription = fetchDatSubscription();
$newSubscription = changeTheSubscriptionModifiersAndBalanceAndAllDatJazz($subscription);
$newSubscription->getRemainingValue() !== $subscription->getRemainingValue() // true;
```

Reply:
Moved back into the Mapper and used an anonymous class to make the calculation work in the test. :handshake: